src/network/gameclient.cc

```diff
@@ -840,8 +840,15 @@ void GameClient::handle_setting_tribes(RecvPacket& packet) {
 			std::string const initialization_script = packet.string();
 			std::unique_ptr<LuaTable> t = lua.run_script(initialization_script);
 			t->do_not_warn_about_unaccessed_keys();
+			std::set<std::string> tags;
+			if (t->has_key("map_tags")) {
+				std::unique_ptr<LuaTable> tt = t->get_table("map_tags");
+				for (int key : tt->keys<int>()) {
+					tags.insert(tt->get_string(key));
+				}
+			}
 			info.initializations.push_back(Widelands::TribeBasicInfo::Initialization(
-			   initialization_script, t->get_string("descname"), t->get_string("tooltip")));
+			   initialization_script, t->get_string("descname"), t->get_string("tooltip"), tags));
```

Comment:
The tags depend on the map, not on the tribe. TribeBasicInfo is the wrong place for them.

Reply:
`tags` is the set of map tags required for the starting condition to be available. It is stored in `info.initializations` so it is available whenever the map changes, to check whether the SC should be shown for this map or not.

Reply:
You're right, I overlooked that.